libraries/botbuilder/tests/streaming/botFrameworkAdapterStreaming.test.js

```diff
@@ -44,12 +51,52 @@ describe('BotFrameworkAdapter Streaming tests', () => {
     it('starts and stops a namedpipe server', () => {
         const adapter = new BotFrameworkAdapter();
 
-        adapter.useNamedPipe('PipeyMcPipeface', async (context) => {
-            await bot.run(context);
+        adapter.useNamedPipe('PipeyMcPipeface', async (_) => {
+            // no op
         });
         expect(adapter.streamingServer.disconnect()).to.not.throw;
     });
 
+    it(`throws exception when trying to connect to a different named pipe than it's connected to`, async () => {
+        const adapter = new ConnectionTestAdapter();
+
+        try {
+            // Don't await to try and connect at the same time
+            await adapter.useNamedPipe(async (_) => {
+                // no op
+            }, 'NamedPipeTest');
+
+            // Try use same NamedPipe again to trigger error.
+            await adapter.useNamedPipe(async (_) => {
```

Comment:
`assert.rejects` IMO is a better way to ensure that a piece of code rejects. You can do:

```typescript
await assert.rejects(adapter.useNamedPipe(...), {
  message: 'This BotFrameworkAdapter ...'
});
```

Reply:
Perfect, thanks for the feedback!